src/fsharp/FSharp.Core/array.fs

```diff
@@ -478,6 +478,17 @@ namespace Microsoft.FSharp.Collections
         [<CompiledName("Where")>]
         let where f (array: _[]) = filter f array
 
+        [<CompiledName("Except")>]
```

Comment:
I went with array1/array2 as that is what is used elsewhere in the module, I agree it could be more meaningful, but that should be clearer from the code.  The name used externally is set in array.fsi, where I think they are called first/second